core/trino-main/src/main/java/io/trino/execution/scheduler/FaultTolerantStageScheduler.java

```diff
@@ -113,12 +122,26 @@
     private final Optional<int[]> sourceBucketToPartitionMap;
     private final Optional<BucketNodeMap> sourceBucketNodeMap;
 
+    private final ScheduledExecutorService executor;
+
     @GuardedBy("this")
     private ListenableFuture<Void> blocked = immediateVoidFuture();
 
     @GuardedBy("this")
     private SettableFuture<Void> taskFinishedFuture;
 
+    // todo make configurable
+    private final Duration minSchedulingDelayDuration = Duration.of(2, ChronoUnit.SECONDS);
```

Comment:
Good point. I started adding new properties yesterday - but I think reusing those is fine. It could be more elastic to have separate ones for case when queries with query-level and task-level retries are run in the same cluster - but it is not a production supported case right now.